clients/redis/redis.go

```diff
@@ -0,0 +1,298 @@
+	return nil, fmt.Errorf("QueryContext is not supported for Redis")
+}
+
+func (s *Store) Begin() (*sql.Tx, error) {
+	return nil, fmt.Errorf("transactions are not supported for Redis")
+}
+
+func (s *Store) LoadDataIntoTable(_ context.Context, _ *optimization.TableData, _ *types.DestinationTableConfig, _, _ sqllib.TableIdentifier, _ types.AdditionalSettings, _ bool) error {
+	return fmt.Errorf("LoadDataIntoTable is not supported for Redis")
+}
+
+func (s *Store) Append(ctx context.Context, tableData *optimization.TableData, _ bool) error {
+	// For Redis, append and merge are the same - we always create new records
+	if _, err := s.Merge(ctx, tableData); err != nil {
+		return fmt.Errorf("failed to append: %w", err)
+	}
+	return nil
+}
+
+// Merge writes rows from TableData as individual entries into a Redis Stream.
+// Each entry contains two fields: the CDC event JSON (artieData) and an emitted-at timestamp (artieEmittedAt).
+func (s *Store) Merge(ctx context.Context, tableData *optimization.TableData) (bool, error) {
+	if tableData.ShouldSkipUpdate() {
+		return false, nil
+	}
+
+	tableID := s.IdentifierFor(tableData.TopicConfig().BuildDatabaseAndSchemaPair(), tableData.Name())
+	redisTableID, ok := tableID.(TableIdentifier)
+	if !ok {
+		return false, fmt.Errorf("expected tableID to be a TableIdentifier, got %T", tableID)
+	}
+
+	rows := tableData.Rows()
+	if len(rows) == 0 {
+		return false, nil
+	}
+
+	cols := tableData.ReadOnlyInMemoryCols().ValidColumns()
+	streamKey := redisTableID.StreamKey()
+	recordsWritten := 0
+	pipeline := s.redisClient.Pipeline()
+
+	for _, row := range rows {
+		rowData := make(map[string]any)
+		for _, col := range cols {
+			value, _ := row.GetValue(col.Name())
+			rowData[col.Name()] = value
+		}
+
+		jsonData, err := json.Marshal(rowData)
+		if err != nil {
+			return false, fmt.Errorf("failed to marshal row data: %w", err)
+		}
+
+		slog.Info("Writing stream entry",
+			slog.String("stream", streamKey),
+			slog.Int("jsonDataLen", len(jsonData)),
+		)
+
+		pipeline.XAdd(ctx, &redis.XAddArgs{
+			Stream: streamKey,
+			MaxLen: streamMaxLen,
+			Approx: true,
+			ID:     "*",
+			Values: map[string]interface{}{
+				artieEmittedAtField: time.Now().UTC().Format(time.RFC3339),
+				artieDataField:      string(jsonData),
+			},
+		})
+		recordsWritten++
+	}
+
+	if _, err := pipeline.Exec(ctx); err != nil {
+		return false, fmt.Errorf("failed to execute pipeline: %w", err)
+	}
+
+	slog.Info("Successfully wrote records to Redis Stream",
+		slog.String("stream", streamKey),
+		slog.Int("recordCount", recordsWritten),
+	)
+
+	return true, nil
+}
+
+func (s *Store) IsRetryableError(err error) bool {
+	if err == nil {
+		return false
+	}
+	// Check for standard network errors (connection reset, refused, timeout, etc.)
+
+	if isRetryableNetworkError(err) {
+		return true
+	}
+
+	// Check for Redis-specific retryable errors
+	errMsg := err.Error()
+
+	if strings.Contains(errMsg, "BUSY") ||
```

Comment:
Can we turn this into a function and use redis errors instead of string contains? If not possible, string contains is ok